rules/no-for-loop-extra-variable.js

```diff
@@ -0,0 +1,133 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+
+function checkForLoop(context, node) {
+	if (checkTwoDeclaredVariables(node.init) && checkIteratorType(node, node.update)) {
+		context.report({
+			node,
+			messageId: 'noForLoopExtraVariable',
+			fix(fixer) {
+				return fixForLoop(node, context, fixer);
+			}
+		});
+	}
+}
+
+function checkTwoDeclaredVariables(init) {
+	return init.type === 'VariableDeclaration' && init.declarations.length === 2;
+}
+
+function checkIteratorType(node, update) {
+	let iteratorNode;
```

Comment:
Sorry, I didn't notice that node is used here, if so, maybe better pass `node` to this function